### PR TITLE
Fix mpl circuit drawing duplication in ipykernel 6

### LINKED ISSUE #6889: Duplicated matplotlib figures when running ipykernel 6
### Information

- **Qiskit Terra version**: 0.18.1
- **Python version**: 3.7.8
- **Operating system**: Linux

### What is the current behavior?

When running ipykernel >= 6.0.0 (latest at the time of writing is `ipykernel==6.0.3`) on Jupyter Notebooks (also in Jupyter Lab) displaying any matplotlib figure using Qiskit (a Quantum Circuit or a visualization of an output) the output is displayed twice. This doesn't seem to happen when running an example of matplotlib in the same notebook.

<details>
  <summary>Notebook example with the issue, running ipykernel 6.0.3</summary>
  
</details>

<details>
  <summary>Same notebook example without issues, running ipykernel 5</summary>  
  
</details>

### Steps to reproduce the problem

- On a fresh Jupyter Notebooks environment (you can try an [online environment](https://jupyter.org/try)) install the latest ipykernel by running `pip install ipykernel==6.0.3`
- Build a QuantumCircuit and draw it using `.draw('mpl')`
- See the circuit figure duplicated

### What is the expected behavior?

To display the figures only once.

### Suggested solutions
N/A

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

The latest major version of ipykernel swapped the matplotlib backend to
the new `matplotlib_inline` separate dependency, which registers itself
with a different backend name.

### Details and comments

Fix #6889.

### CODE REVIEW

**Comment by jakelishman** [comment]
This PR should be backported, but won't actually apply because #6675 merged before it, and that's not stable for backport.  It's a trivial one-line change, though, so I'll manually backport.

File: qiskit/visualization/matplotlib.py [hunk 1]
```diff
@@ -350,7 +350,11 @@ def draw(self, filename=None, verbose=False):
         if not self._user_ax:
             from matplotlib import get_backend
 
-            if get_backend() in ["module://ipykernel.pylab.backend_inline", "nbAgg"]:
+            if get_backend() in [
+                "module://ipykernel.pylab.backend_inline",
+                "module://matplotlib_inline.backend_inline",
+                "nbAgg",
+            ]:
```
Comment (mtreinish) [inline qiskit/visualization/matplotlib.py:357]: I think the same if statement exists for all the matplotlib based visualizations, state, histogram, pulse, etc. We probably need to update it in all of them

**Comment by jakelishman** [comment]
There we go - force-pushed rebase finally convinced GH to find the missing commits from my branch (by overwriting them...).  Hopefully this was the last weird hangover from the downtime yesterday.

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for updating all the figure returns. I tested it locally and it works for me with ipykernel>=6.0.0

**Pull request merged**